src/coreclr/tools/Common/TypeSystem/Interop/InteropStateManager.cs

```diff
@@ -446,31 +447,33 @@ public PInvokeDelegateWrapperHashtable(InteropStateManager interopStateManager,
             }
         }
 
-        private sealed class PInvokeLazyFixupFieldHashtable : LockFreeReaderHashtable<MethodDesc, PInvokeLazyFixupField>
+        private readonly record struct PInvokeLazyFixupFieldKey(MethodDesc Method, int SignatureBytes);
```

Comment:
Why is the SignatureBytes part of the key? The number is derived from MethodDesc - it should not need to be part of the key.

Reply:
That’s the part I don’t like but I don’t know how to propagate it into the value in LockFreeHashtable.

Reply:
I made another attempt at this, but it still feels odd. `LockFreeReaderHashtable` is designed to create the value from the key and there's no easy way to pass the additional information into the `PInvokeLazyFixupField` constructor.